lib/mayaUsd/ufe/StagesSubject.cpp

```diff
@@ -243,6 +253,10 @@ void processAttributeChanges(
     if (sendRemoved) {
         attributeChanged(ufePath, changedPath.GetNameToken(), AttributeChangeType::kRemoved);
     }
+    if (MayaUsd::ufe::InAttributeMetadataChange::inAttributeMetadataChange()) {
```

Comment:
I am quite concerned about this approach, as it is very weak.  It depends on going through the maya-usd code path you've defined, and will not work when using e.g. the Pixar USD API alone.  We should move away from using this style of coding, and instead rely on USD notifications themselves, which will be bullet-proof.  USD does provide significant detail about what fields have changed in its notification system; e.g. see
https://github.com/Autodesk/maya-usd/blob/42854b32c9c1cd706917e088b477119ca1d791ab/lib/mayaUsd/ufe/StagesSubject.cpp#L517
I'm very, very hopeful we can use that.

Reply:
Indeed, this forces me to recheck the USD API to see if there is a better way:
On a Metadata change, the function SdfChangeList::DidChangeInfo() will be called, generating a very specifically shaped USD notification that should be intercepted instead of introducing a Metadata guard.
This means that looking for contents in the `.infoChanged` field of the entry should be enough to detect that metadata was modified.
Also means that the UFE API for AttributeMetadataChanged requires a list of metadata keys to copy all those found there.

Reply:
I hadn't wanted to implement things this way, but I was under the impression that there was no way to find out that USD had made a metadata change. But now Jerry bellow is suggesting it is possible so I'll look into it.